quinn/src/platform/unix.rs

```diff
@@ -265,11 +266,15 @@ impl super::UdpExt for UdpSocket {
         meta[0] = decode_recv(&name, &hdr, n as usize);
         Ok(1)
     }
+
+    fn caps() -> super::UdpCapabilities {
+        *CAPABILITIES
+    }
 }
 
 const CMSG_LEN: usize = 64;
 
-fn prepare_msg(
+fn prepare_msg<U: super::UdpExt>(
```

Comment:
If `caps()` became a free function in `platform` (e.g. `pub use imp::udp_caps;`) we could avoid the generic here.

Reply:
Actually, if we want to commit to the global state, we could even just have a `static CAPABILITIES` exported directly from `platform`.

Reply:
I agree, it's not strictly necessary to be on this trait. I can however see one reason why it might stay there: In case there are multiple IO providers for a single platform (e.g. lets say one also implements raw socket support or XDP support for Linux).  
But maybe that should then be in a different platform module?

Regarding function vs public static ref: Static ref seems easier. Doesn't however allow to e.g. have a `const` impl on platforms which don't require runtime checks. But the difference between those seems totally negligible given it's a 1-time thing. The other difference I can see is that with the static ref this detail leaks out of the module.

Reply:
On consideration, I'm weakly in favor of a function since it's totally opaque that way, bearing in mind that we shouldn't have a `const` impl on any platform because that would constitute a cfg-dependent signature which is a huge landmine.

Multiple dynamically-selectable backends on a single platform are an interesting idea, but aren't something we currently support whatsoever, and I don't think this would have a meaningful impact on the amount of work needed to make that happen.

Reply:
I switched things for function